doc/manual/typing.md

```diff
@@ -430,13 +434,134 @@ Because the `match` statement has a catch-all case `_`, this function is indeed
 able to handle other tags than `http` and `ftp`, as expressed by its polymorphic
 type.
 
+#### Enum row polymorphism
+
+Row polymorphism also works with enum types. As for records, a polymorphic enum
+tail means that said tail can be substituted for any other enum type, indicating
+an enum type that can be extended arbitrarily.
+
+This typically happens when a match expression has a catch-all case:
+
+```nickel #repl
+> let is_ok : forall a b tail. [| 'Ok a, 'ok b ; tail |] -> Bool =
+    match {
+      'Ok x => true,
+      'ok x => true,
+      _ => false,
+    }
+  in
+  is_ok 'other
+false
+```
+
+In this example, the match expression can handle any enum value thanks to the
+presence of the catch-all case `_ => false`. This is reflected in the argument
+type of `is_ok` by the polymorphic tail `; tail`.
+
+A more advanced usage of row polymorphism for enum types is widening. Widening
+is a way of making an enum type "embeddable" in larger enum types, so to speak.
+Most of the time, you don't have to think about it, thanks to the way Nickel
+infers enum types. Take the following example:
+
+```nickel #parse
+(
+  let foo : [| 'Foo Number |] = 'Foo 5 in
+  foo |> match {
+    'Foo x => x,
+    'Bar x => x,
+  }
+) : _
+```
+
+This example looks entirely legit, but if you try to run it, you'll get the
+following error:
+
+```text
+error: type error: missing row `Bar`
+  ┌─ <repl-input-2>:3:3
+  │
+3 │   foo |> match {
+  │   ^^^ this expression
+  │
+  = Expected an expression of type `[| 'Foo Number, 'Bar Number |]`, which contains the field `Bar`
+  = Found an expression of type `[| 'Foo Number |]`, which does not contain the field `Bar`
+```
+
+The match expression expects a type `[| 'Foo Number, 'Bar Number |]`, but `foo`
+is of type `[| 'Foo Number |]`. However, `[| 'Foo Number |]` *should be
+compatible* with `[| 'Foo Number, 'Bar Number |]`: if something is an enum `'Foo
+Number`, then it is surely either `'Foo Number` or `'Bar Number`. This
```

Comment:
Do the backticks around `'Foo Number` render correctly, even though there's a newline?

Reply:
I _think_ so, because it happens often when I use neovim to wrap lines around automatically. But let me double check